Library/Homebrew/brew.rb

```diff
@@ -77,14 +75,15 @@
     # `Homebrew.help` never returns, except for external/unknown commands.
   end
 
-  # Migrate LinkedKegs/PinnedKegs if update didn't already do so
-  UpdateMigrator.migrate_legacy_keg_symlinks_if_necessary
-
   # Uninstall old brew-cask if it's still around; we just use the tap now.
   if cmd == "cask" && (HOMEBREW_CELLAR/"brew-cask").exist?
     system(HOMEBREW_BREW_FILE, "uninstall", "--force", "brew-cask")
   end
 
+  if ENV["HOMEBREW_BUILD_FROM_SOURCE"]
+    odeprecated("HOMEBREW_BUILD_FROM_SOURCE", "--build-from-source")
+  end
```

Comment:
Please, please tell me that this variable and cmdline flag are not going away.

Reply:
@chdiza @RandomDSdevel Yes, the `HOMEBREW_BUILD_FROM_SOURCE` environment variable is going away. It tells people to not file issues they encounter but they keep doing so regardless. This is a waste of time for the Homebrew maintainers.

Homebrew is not a from-source package manager any more. https://github.com/Homebrew/brew/pull/5491 means that building from source is going to generate identical output to bottles. https://github.com/Homebrew/homebrew-core/issues/31510 means there will be no reason to build from source when using the supported Homebrew installation location. `--build-from-source` is sticking around for people to be able to reproduce issues by building specific packages from source on request.

Reply:
@MikeMcQuaid I apologize that people report issues about source builds. I did that once, before I understood that they weren't officially supported. However, building from source is an extremely valuable feature to me. Allowing users to set `HOMEBREW_BUILD_FROM_SOURCE` feels right. Personally, I *love* digging through the source code of installed programs and libraries, and seeing their `make` invocations go flying by. **In fact, it was this easy accessibility of local source code that pulled me into the open-source movement** (I maintain the FlatBuffers project as a volunteer). Furthermore, it gives me peace of mind in the unfortunate event that there are more security issues with Bottle creations: building from source decentralizes the creation of binary artifacts. It's clear that you spend a **lot** of time maintaining this project, and for that we are thankful. If you would please consider keeping `HOMEBREW_BUILD_FROM_SOURCE` available, deeply-technical and curious enthusiasts like myself will continue to love this project.

Reply:
@rw You can still dig through the source code with `brew install --interactive` or `brew install --build-from-source`. You cannot globally tell Homebrew to build everything from source as it's a waste of both your time and our time when people file issues. Additionally, (and I'm not actually joking here) it's also a waste of energy to compile everything repeatedly unnecessarily. 

We are not bringing back this variable, sorry.